src/helpers/filling.rs

```diff
@@ -97,3 +100,78 @@ pub fn fill_tilemap_rect_color(
         }
     }
 }
+
+/// Generates a vector of hex positions that form a ring of given `radius` around the specified
+/// `origin`.
+///
+/// If `radius` is zero, `origin` is the only position in the returned vector.
+pub fn generate_hex_ring(origin: AxialPos, radius: u32) -> Vec<AxialPos> {
+    if radius == 0 {
+        vec![origin]
+    } else {
+        let mut ring = Vec::with_capacity((radius * 6) as usize);
+        let corners = HEX_DIRECTIONS
+            .iter()
+            .map(|direction| origin + radius * AxialPos::from(direction))
+            .collect::<Vec<AxialPos>>();
+        // The "tangent" is the direction we must travel in to reach the next corner
+        let tangents = (0..6)
+            .map(|ix| HexDirection::from(ix + 2).into())
+            .collect::<Vec<AxialPos>>();
+
+        for (&corner, &tangent) in corners.iter().zip(tangents.iter()) {
+            for k in 0..radius {
+                ring.push(corner + k * tangent);
+            }
+        }
+
+        ring
+    }
+}
+
+/// Generates a vector of hex positions that form a hexagon of given `radius` around the specified
+/// `origin`.
+pub fn generate_hexagon(origin: AxialPos, radius: u32) -> Vec<AxialPos> {
+    let mut hexagon = Vec::with_capacity((6 * radius * (radius + 1) / 2) as usize);
+    for r in 0..radius {
+        hexagon.extend(generate_hex_ring(origin, r));
+    }
+    hexagon
+}
+
+/// Fills a hexagonal region with the given `tile_texture`.
+///
+/// The rectangular region is defined by an `origin` in [`TilePos`](crate::tiles::TilePos), and a
+/// `radius`.
+///
+/// Tiles that do not fit in the tilemap will not be created.
+pub fn fill_tilemap_hexagon(
+    tile_texture: TileTexture,
+    origin: TilePos,
+    radius: u32,
+    hex_coord_system: HexCoordSystem,
+    tilemap_id: TilemapId,
+    commands: &mut Commands,
+    tile_storage: &mut TileStorage,
+) {
+    let tile_positions = generate_hexagon(
+        AxialPos::from_tile_pos_given_coord_system(&origin, hex_coord_system),
+        radius,
+    )
+    .into_iter()
+    .map(|axial_pos| axial_pos.as_tile_pos_given_coord_system(hex_coord_system))
+    .collect::<Vec<TilePos>>();
+
+    for tile_pos in tile_positions {
+        let tile_entity = commands
```

Comment:
This will be measurably faster with spawn_batch. Probably want to change the sibling to match? Or just open an issue over it.

Reply:
See: https://github.com/StarArawn/bevy_ecs_tilemap/issues/300